osu.Game/Overlays/Dialog/PopupDialogDangerousButton.cs

```diff
@@ -92,9 +94,28 @@ protected override void OnMouseUp(MouseUpEvent e)
                 {
                     lowPassFilter.CutoffTo(AudioFilter.MAX_LOWPASS_CUTOFF);
                     AbortConfirm();
+                    mouseDown = false;
                 }
             }
 
+            protected override bool OnHover(HoverEvent e)
+            {
+                if (mouseDown)
+                    BeginConfirm();
+
+                return base.OnHover(e);
+            }
+
+            protected override void OnHoverLost(HoverLostEvent e)
+            {
+                base.OnHoverLost(e);
+
+                if (!mouseDown) return;
+
+                lowPassFilter.CutoffTo(AudioFilter.MAX_LOWPASS_CUTOFF);
```

Comment:
This is copied verbatim from `OnMouseUp()`; I'd argue that it shouldn't be, and it should instead be part of `AbortConfirm()` as I don't see a case in this class where you'd want to abort *without* sweeping the filter. Requires making it virtual, but I think it makes sense. Or you could split a local private method, although that is a little bit more confusing perhaps.